pyarlo/camera.py

```diff
@@ -132,95 +135,112 @@ def get_battery_level(self):
         return base.get_camera_battery_level[self.device_id]
 
     @property
-    def get_signal_strength(self):
-        """Get the camera Signal strength."""
+    def properties(self):
+        """Get this camera's extended properties."""
         base = self._session.base_stations[0]
         props = base.get_camera_properties
         if not props:
             return None
+
         for cam in props:
-            if cam['serialNumber'] == self.device_id:
-                return cam['signalStrength']
+            if cam["serialNumber"] == self.device_id:
+                return cam
```

Comment:
this code to load a camera's properties was duplicated a lot, so i just made a `self.properties` property that loaded up the ... properties. 😄 

this simplifies a lot of the lower properties and methods.

Reply:
@viswa-swami  how does this change looks for you? Will it break the HA component?

Reply:
For HASS, it would be better if we return the SignalStrength directly.  The reason is HASS owners tend to avoid accessing any component specific parameters inside HASS.  In this case, we would have to filter by 'SignalStrength' or similarly any other parameter inside cam object if required.

Reply:
The diff here is confusing... `ArloCamera.get_signal_strength` still exists and is now covered with a unittest, I simply broke the property loading/parsing part of it out into a separate method since the code was reused so many times.